services/vision/mlvision/detector.go

```diff
@@ -47,42 +49,42 @@ func attemptToBuildDetector(mlm mlmodel.Service) (objectdetection.Detector, erro
 	return func(ctx context.Context, img image.Image) ([]objectdetection.Detection, error) {
 		origW, origH := img.Bounds().Dx(), img.Bounds().Dy()
 		resized := resize.Resize(inWidth, inHeight, img, resize.Bilinear)
-		inMap := make(map[string]interface{})
+		inMap := ml.Tensors{}
 		switch inType {
 		case UInt8:
-			inMap["image"] = rimage.ImageToUInt8Buffer(resized)
+			inMap["image"] = tensor.New(
+				tensor.WithShape(1, int(inHeight), int(inWidth), 3),
```

Comment:
I think rather than using `inHeight` and `inWidth` it would be better to query `resized.Bounds.Dx()` and `resized.Bounds.Dy()` for the shape. That way, when https://github.com/viamrobotics/rdk/pull/2809 lands, it will just work in the case where you have -1's for `inHeight` and `inWidth`. Here and elsewhere.

Reply:
good point, I added your upstream changes and made the shape use resizeH and resizeW